### PR TITLE
BEI-10: setup code coverage with jest

### PR DESCRIPTION
**Pull request by LucianaDera**
WHY THIS PR?

We need a test to fail if code coverage does not reach the required threshold of 60% on all metrics.

WHAT DOES THE PR DO?
- It can help detect bugs early in the development process.
- Serves as a metric for assessing the quality of the codebase.
- test will fail if it does not reach the required threshold.

HOW CAN THIS PR BE TESTED?
- Configure test coverage with jest.
- Run test.

THINGS YOU MAY WANT THE REVIEWER TO NOTE.
N/A

### CODE REVIEW

File: .hound.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
 eslint:
   enabled: true
   config_file: .eslintrc
-fail_on_violations: true
+  fail_on_violations: true
```
Comment (LordUche) [inline .hound.yml:4]:
```suggestion
fail_on_violations: true
```

**Review by LordUche — CHANGES_REQUESTED** [review]

File: .jest.config.js
Comment (RICHYDOS) [inline .jest.config.js:1]: Is the file name meant to be ".jest.config.js" or "jest.config.js"

**Review by LordUche — CHANGES_REQUESTED** [review]

**Comment by hizikex** [comment]
Resolve the branch conflicts.

**Review by hizikex — APPROVED** [review]

**Review by RICHYDOS — APPROVED** [review]
Well Done

File: tsconfig.json [hunk 2]
```diff
@@ -12,9 +12,9 @@
...
-    "strict": true,
-    "noUnusedLocals": true,
-    "noImplicitReturns": true,
-    "skipLibCheck": true
-  }
+    "strict": true, 
+    "noUnusedLocals": true, 
+    "noImplicitReturns": true, 
+    "skipLibCheck": true  
+  },
```
Comment (LordUche) [inline tsconfig.json:19]: These changes have nothing to do with the task

File: package.json [hunk 3]
```diff
@@ -14,9 +15,10 @@
   "author": "",
   "license": "ISC",
   "devDependencies": {
+    "jest": "^29.6.1",
+    "jest-cli": "^29.6.1",
```
Comment (LordUche) [inline package.json:19]: Why do you need this package?

**Comment by LordUche** [comment]
@LucianaDera Please add `coverage/` to `.gitignore`

**Review by LordUche — CHANGES_REQUESTED** [review]
Great job 👍
Please pull the latest changes on staging and resolve all conflicts.

**Review by LordUche — APPROVED** [review]

**Pull request merged**